server/src/main/java/com/google/coffeehouse/common/Person.java

```diff
@@ -37,6 +47,41 @@ private Person(Builder builder) {
     this.userId = builder.idGenerator.generateId();
   }
 
+  /** Overloaded static factory, calls other fromMap with null IdentifierGenerator. */
+  public static Person fromMap(Map personInfo) {
+    return Person.fromMap(personInfo, null);
+  }
+
+  /** 
+   * Creates a {@link Person} object from a Map with the relevant parameters as keys.
+   * @param personInfo the Map that contains the information used to construct the Person. At a
+   *     minimum this includes an {@code "email"} key that is mapped to a String that describes
+   *     the email of the Person to be created, as well as a {@code "nickname"} key that is mapped
+   *     to a String that describes the name of the Person to be created. A {@code "pronouns"} key
+   *     that is mapped to a String describing the pronouns of the Person can optionally be added.
+   * @param idGen the {@link IdentifierGenerator} used when constructing the Person. If
+   *     null, the default generator will be used
+   * @return the created Person
+   * @throws IllegalArgumentException if no valid {@code "email"} key or 
+   *     {@code "nickname"} key is defined
+   */
```

Comment:
Wanted to call out that your Java Docs documentation is super thorough and well done, makes life easier on my end